### PR TITLE
feat(SDK-4183): Event properties normalisation

### PR DESCRIPTION
**Pull request by CTLalit**
- backend sends normalised event properties, we need to compare on sdk side based on normalised event which client app might fire

### CODE REVIEW

File: clevertap-core/src/main/java/com/clevertap/android/sdk/variables/JsonUtil.java [hunk 1]
```diff
@@ -68,7 +74,9 @@ public static <T> Map<String, T> mapFromJson(JSONObject object) {
         value = null;
       }
       T castedValue = uncheckedCast(value);
-      result.put(key, castedValue);
+      if (normaliseNames) {
+        result.put(Utils.getNormalizedName(key), castedValue);
+      }
```
Comment (vasct) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/variables/JsonUtil.java:79]: Missing the else statement for when normaliseNames=false

File: clevertap-core/src/main/java/com/clevertap/android/sdk/variables/JsonUtil.java [hunk 2]
```diff
@@ -50,6 +52,10 @@ public static JSONArray listToJsonArray(Iterable<?> list) throws JSONException {
   }
 
   public static <T> Map<String, T> mapFromJson(JSONObject object) {
+    return mapFromJson(object, false);
+  }
+
+  public static <T> Map<String, T> mapFromJson(JSONObject object, boolean normaliseNames) {
     if (object == null) {
       return null;
     }
```
Comment (vasct) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/variables/JsonUtil.java:61]: keysIterator is actually of type `Iterator<String>`. If it is changed the cast of `key` to (String) will be redundant.

**Review by vasct — CHANGES_REQUESTED** [review]

File: clevertap-core/src/main/java/com/clevertap/android/sdk/variables/JsonUtil.java (hunk 2 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/variables/JsonUtil.java:61]: done.

**Review by vasct — DISMISSED** [review]

**Comment by vasct** [comment]
@CTLalit check the failing tests.

**Comment by vasct** [comment]
Also it could be an easier implementation if we only normalize the names when comparing the properties instead of always getting them normalized. This is what I did for the event names and profile properties.

**Pull request closed without merging**